app/src/main/java/com/randomappsinc/studentpicker/home/NameListsAdapter.java

```diff
@@ -23,15 +23,17 @@
 import butterknife.ButterKnife;
 import butterknife.OnClick;
 
-public class NameListsAdapter extends BaseAdapter {
+public class NameListsAdapter extends RecyclerView.Adapter<NameListsAdapter.NameListViewHolder> {
 
     private Context context;
```

Comment:
Having adapters store a reference to a context is a legacy way of doing things when I frankly didn't know any better, haha. It's dangerous to store contexts because they are often times activities and if you don't clean up references to them properly, you leak an entire activity, which is massive (you saw me do this in TravelBuddy when you ran LeakCanary on it, haha). Given that we're doing a refactor, let's take the opportunity to leave the code in a better place than which we found it and remove this reference and the parameter it adds in the constructor.